tracer/test/test-applications/integrations/Samples.RabbitMQ/Program.cs

```diff
@@ -220,23 +223,26 @@ private static void Receive(bool useQueue)
 
                 if (useQueue)
                 {
-                    DequeueThread.Start();
+                    _dequeueThread = new Thread(() => ConsumeFromQueue(queue));
+                    _dequeueThread.Start();
                 }
 
-                while (_messageCount != 0)
+                while (Interlocked.Read(ref _messageCount) != 0)
                 {
                     Thread.Sleep(100);
                 }
 
+                queue.CompleteAdding();
+
                 Console.WriteLine("[Receive] Exiting Thread.");
             }
         }
 
-        private static void ConsumeFromQueue()
+        private static void ConsumeFromQueue(BlockingCollection<BasicDeliverEventArgs> queue)
         {
-            while (_queue.Count > 0 )
+            while (!queue.IsCompleted)
```

Comment:
```
foreach (var ea in queue.GetConsumingEnumerable())
{
    TraceOnTheReceivingEnd(ea);
}

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2736/commits/264059a6144006ce866b77bade9246bdf8910806